### PR TITLE
null-annotate lib/ui/hooks.dart

### PR DESCRIPTION
**Pull request by cbracken**

### CODE REVIEW

File: lib/ui/hooks.dart [hunk 1]
```diff
@@ -269,7 +269,7 @@ void _invoke(void callback(), Zone zone) {
 }
 
 /// Invokes [callback] inside the given [zone] passing it [arg].
-void _invoke1<A>(void callback(A a), Zone zone, A arg) {
+void _invoke1<A>(void callback(A a)/*?*/, Zone/*!*/ zone, A arg) {
```
Comment (cbracken) [inline lib/ui/hooks.dart:272]: Thoughts on whether we should mark the callback params one way or another? Not entirely sure I trust the migration tool's judgement on the `_invoke*` calls.

Comment (yjbanov) [reply lib/ui/hooks.dart:272]: Good point. I think we should. These are basically anonymous `typedef`s, and I would definitely annotate a `typedef`.

File: lib/ui/hooks.dart [hunk 2]
```diff
@@ -65,11 +65,11 @@ String _localeClosure() {
 
 @pragma('vm:entry-point')
 // ignore: unused_element
-_LocaleClosure _getLocaleClosure() => _localeClosure;
+_LocaleClosure/*?*/ _getLocaleClosure() => _localeClosure;
```
Comment (yjbanov) [inline lib/ui/hooks.dart:68]: I think this is safe to be non-null. `_localeClosure` is just a tear-off of a top-level function.

File: lib/ui/hooks.dart [hunk 3]
```diff
@@ -65,11 +65,11 @@ String _localeClosure() {
...
 
 @pragma('vm:entry-point')
 // ignore: unused_element
-void _updateLocales(List<String> locales) {
+void _updateLocales(List<String/*!*/>/*!*/ locales) {
```
Comment (yjbanov) [inline lib/ui/hooks.dart:72]: Elzar approves

**Pull request merged**